### PR TITLE
allows progress integration tests to work with different regexes

### PR DESCRIPTION
**Pull request by shamsimam**

### CODE REVIEW

File: integration/tests/cook/util.py [hunk 1]
```diff
@@ -467,3 +467,13 @@ def sleep_for_publish_interval(cook_url):
     progress_publish_interval_ms = get_in(cook_settings, 'progress', 'publish-interval-ms')
     wait_publish_interval_ms = min(3 * progress_publish_interval_ms, 20000)
     time.sleep(wait_publish_interval_ms / 1000.0)
+
+def output_progress_string(cook_url, percent, message):
```
Comment (dposada) [inline integration/tests/cook/util.py]: Maybe just `progress_line`? The "output" is kind of confusing

File: integration/tests/cook/test_basic.py [hunk 2]
```diff
@@ -137,7 +137,8 @@ def test_failing_submit(self):
 
     def test_progress_update_submit(self):
         job_executor_type = util.get_job_executor_type(self.cook_url)
-        command = 'echo "progress: 25 Twenty-five percent in ${PROGRESS_FILE}" >> ${PROGRESS_FILE}; sleep 1; exit 0'
+        m1 = util.output_progress_string(self.cook_url, 25, 'Twenty-five percent in ${PROGRESS_FILE}')
```
Comment (dposada) [inline integration/tests/cook/test_basic.py]: Maybe `line` instead of `m1`

File: integration/tests/cook/test_basic.py [hunk 3]
```diff
@@ -200,12 +201,14 @@ def test_configurable_progress_update_submit(self):
...
-                  'echo "progress: 50 Fifty percent" && sleep 2 && ' \
-                  'echo "progress: Sixty percent invalid format" && sleep 2 && ' \
-                  'echo "progress: 75 Seventy-five percent" && sleep 2 && ' \
-                  'echo "progress: Eighty percent invalid format" && sleep 2 && ' \
-                  'echo "Done" && exit 0'
+        m1 = util.output_progress_string(self.cook_url, 25, 'Twenty-five percent')
+        m2 = util.output_progress_string(self.cook_url, 50, 'Fifty percent')
+        m3 = util.output_progress_string(self.cook_url, '', 'Sixty percent invalid format')
+        m4 = util.output_progress_string(self.cook_url, 75, 'Seventy-five percent')
+        m5 = util.output_progress_string(self.cook_url, '', 'Eighty percent invalid format')
```
Comment (dposada) [inline integration/tests/cook/test_basic.py]: Maybe `line_X` instead of `mX`

File: integration/tests/cook/test_basic.py [hunk 4]
```diff
@@ -232,10 +235,10 @@ def test_multiple_progress_updates_submit(self):
 
     def test_multiple_rapid_progress_updates_submit(self):
         job_executor_type = util.get_job_executor_type(self.cook_url)
-        command = ''.join(['echo "progress: {0} {0}%" && '.format(a) for a in range(1, 100, 4)]) + \
-                  ''.join(['echo "progress: {0} {0}%" && '.format(a) for a in range(99, 40, -4)]) + \
-                  ''.join(['echo "progress: {0} {0}%" && '.format(a) for a in range(40, 81, 2)]) + \
-                  'echo "Done" && exit 0'
+        def progress_string(a):
+            return util.output_progress_string(self.cook_url, a, f'{a}%')
```
Comment (dposada) [inline integration/tests/cook/test_basic.py]: I would favor a blank line before and after the `def`

File: integration/tests/cook/test_basic.py (hunk 2 above)
Comment (shamsimam) [reply integration/tests/cook/test_basic.py]: Done.

File: integration/tests/cook/util.py (hunk 1 above)
Comment (shamsimam) [reply integration/tests/cook/util.py]: Done.

File: integration/tests/cook/test_basic.py (hunk 4 above)
Comment (shamsimam) [reply integration/tests/cook/test_basic.py]: Done.

File: integration/tests/cook/test_basic.py (hunk 3 above)
Comment (shamsimam) [reply integration/tests/cook/test_basic.py]: Done.

**Comment by shamsimam** [comment]
@dposada build is green now

**Pull request merged**